src/ng-select/selection-model.ts

```diff
@@ -30,6 +31,7 @@ export class DefaultSelectionModel implements SelectionModel {
                 const childrenCount = item.parent.children.length;
                 const selectedCount = item.parent.children.filter(x => x.selected).length;
                 item.parent.selected = childrenCount === selectedCount;
+                item.parent.indeterminate = selectedCount > 0 && !item.parent.selected ? true : false;
```

Comment:
selectedCount will always be > 0 here since you are selecting child item. Can be simply one check:
```ts
item.parent.indeterminate = !item.parent.selected;
```